mne/decoding/transformer.py

```diff
@@ -245,6 +245,54 @@ def inverse_transform(self, X, y=None):
         return X.reshape(-1, self.n_channels, self.n_times)
 
 
+class Vectorizer(object):
+    """Mixin class for transformation of different data to fit into
+       scikit-learn pipeline.
+    """
+
+    def fit(self, data, y=None):
+        """Fit data and parameters into self. This is defined for the sake
+           of completeness of scikit-learn API.
+        
+        Parameters
+        ----------
+        data : array, 2D or 3D matrix
+        y : None or array of labels
+
+        Returns
+        -------
+        self : instance of VectorizerMixin
+        """
+        return self
+
+    def transform(self, data, y=None):
+        """Transform data matrix into 2D matrix. Data could be of
+           Epochs, Evoked, Raw, Covariance.
+
+        Parameters
+        ----------
+        data : array, 2D or 3D matrix
+        y : None or array of labels
+
+        Returns
+        -------
+        X : array, 2D matrix
+        """
+        import numpy as np
+
+        #TODO: Make it work for Evoked, Raw, Covariance
+        if not isinstance(data, np.ndarray):
+            raise ValueError("data should be ndarray (got %s)."
+                             % type(data))
+
+        if data.ndim == 3:
+            n_epochs, n_channels, n_times = data.shape
+            return data.reshape(n_epochs, n_channels * n_times)
+        
+        elif data.ndim == 2:
```

Comment:
This is how I expect the code to be. I am looking for everyone's views on how to proceed and will being the actual coding once everything is finalized. :)